[iOS] Increase the limit on the canvas size to 8192x8192

shallawa opened the pull request:
#### d1f63c061eadee6c83dc9fa06a2725c3d099a86b
<pre>
[iOS] Increase the limit on the canvas size to 8192x8192
<a href="https://bugs.webkit.org/show_bug.cgi?id=271002">https://bugs.webkit.org/show_bug.cgi?id=271002</a>
<a href="https://rdar.apple.com/122230873">rdar://122230873</a>

Reviewed by Chris Dumez.

The maximum size of the canvas on iOS has been 4096x4096. The jetsam limit on iOS
has been raised. So we can safely double the limit on the canvas size on iOS.

* LayoutTests/fast/canvas/canvas-size-minimum-limit-expected.txt: Added.
* LayoutTests/fast/canvas/canvas-size-minimum-limit.html: Added.
* LayoutTests/platform/ios/compositing/canvas/accelerated-canvas-compositing-size-limit-expected.txt:
* LayoutTests/platform/ios/fast/canvas/canvas-toDataURL-crash-expected.txt:
* LayoutTests/platform/ios/fast/canvas/image-buffer-backend-variants-expected.txt:
* LayoutTests/platform/ios/fast/canvas/offscreen-giant-transfer-to-imagebitmap-expected.txt:
* LayoutTests/platform/ios/fast/canvas/pattern-too-large-to-create-expected.txt:
* LayoutTests/platform/ios/fast/canvas/resize-to-large-canvas-and-convert-to-blog-expected.txt:
* Source/WebCore/html/CanvasBase.cpp:
(WebCore::maxCanvasArea):

Canonical link: <a href="https://commits.webkit.org/276145@main">https://commits.webkit.org/276145@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f56c14a410952945abb1529ba7df89a75435e26e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43883 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22929 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46306 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46521 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39960 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46186 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26858 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20327 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36204 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44457 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19991 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37783 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17206 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38862 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1933 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40077 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39162 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48082 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18887 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15461 "Found 1 new test failure: http/tests/navigation/parsed-iframe-dynamic-form-back-entry.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43017 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20280 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/38051 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41719 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9763 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20484 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19904 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on LayoutTests/fast/canvas/canvas-size-minimum-limit.html:
```diff
@@ -0,0 +1,23 @@
+<head>
+    <script src="../../resources/js-test-pre.js"></script>
```
Please use `js-test.js`. Your test output is incomplete at the moment because you're using pre.js without post.js.

shallawa committed:
[iOS] Increase the limit on the canvas size to 8192x8192
https://bugs.webkit.org/show_bug.cgi?id=271002
rdar://122230873

Reviewed by Chris Dumez.

The maximum size of the canvas on iOS has been 4096x4096. The jetsam limit on iOS
has been raised. So we can safely double the limit on the canvas size on iOS.

* LayoutTests/fast/canvas/canvas-size-minimum-limit-expected.txt: Added.
* LayoutTests/fast/canvas/canvas-size-minimum-limit.html: Added.
* LayoutTests/platform/ios/compositing/canvas/accelerated-canvas-compositing-size-limit-expected.txt:
* LayoutTests/platform/ios/fast/canvas/canvas-toDataURL-crash-expected.txt:
* LayoutTests/platform/ios/fast/canvas/image-buffer-backend-variants-expected.txt:
* LayoutTests/platform/ios/fast/canvas/offscreen-giant-transfer-to-imagebitmap-expected.txt:
* LayoutTests/platform/ios/fast/canvas/pattern-too-large-to-create-expected.txt:
* LayoutTests/platform/ios/fast/canvas/resize-to-large-canvas-and-convert-to-blog-expected.txt:
* Source/WebCore/html/CanvasBase.cpp:
(WebCore::maxCanvasArea):

Canonical link: https://commits.webkit.org/276145@main
